index.js

```diff
@@ -5,23 +5,28 @@ import fs from "fs";
 const drugs = [
   new Drug("Doliprane", 20, 30),
   new Drug("Herbal Tea", 10, 5),
-  new Drug("Fervex", 5, 40),
+  new Drug("Fervex", 12, 35),
   new Drug("Magic Pill", 15, 40)
 ];
 const trial = new Pharmacy(drugs);
 
 const log = [];
 
 for (let elapsedDays = 0; elapsedDays < 30; elapsedDays++) {
-  log.push(JSON.stringify(trial.updateBenefitValue()));
+  log.push(JSON.parse(JSON.stringify(trial.updateBenefitValue())));
```

Comment:
```suggestion
  log.push(trial.updateBenefitValue());
```
It does not work like this?

Reply:
By the way, why is it called a `trial`? I suggest we rename the variable `pharmacy`

Reply:
Unfortunately no it doesn't: it will log me 30 times the last result of `trial.updateBenefitValue()`. That's why I ended up with this parse/stringify (I couldn't find another solution quickly, I'll have another look!)

Reply:
I understand why. Since the code mutated `this.drugs`, without you code, you always have the same reference (mutated little by little) in the array.
I think your code is pragmatic then

Reply:
(sorry I missed your comment on `trial` I'll rename it now!)